lib/Doctrine/ORM/Query.php

```diff
@@ -146,9 +146,9 @@ final class Query extends AbstractQuery
     /**
      * The first result to return (the "offset").
      *
-     * @var int|null
+     * @var int
      */
-    private $firstResult = null;
+    private $firstResult = 0;
```

Comment:
This will require a deprecation of setFirstResult(null). The modifyLimitQuery related changes rhis enables are nice!

Reply:
I'd say it's a valid logical step but not a requirement. From the standpoint of the end result, `0` and `null` here are and have always been identical, so this patch doesn't change anything about that.

Reply:
This is actually a breaking change. Symfony's test suite is now failing because Symfony's `ORMQueryBuilderLoader` checks if `getFirstResult()` returns `null`.

Reply:
What part of the API contract does this change break?

Reply:
It breaks code that does not treat `0` and `null` as equal. One could argue though that such a piece of code has always been broken or fragile at least.

Here's the fix for Symfony: symfony/symfony#46282

Reply:
No, I specifically mean the API contract. If passed null, the component will do this. If passed a zero, it will do that. What has changed?

Reply:
The return value of `$em->createQueryBuilder()->getFirstResult()` has changed from `null` to `0`. Do we consider this to be part of the API contract? The doc block says we do:

https://github.com/doctrine/orm/blob/07ee5552793e618ac2e3ac9f3541b3749238b8e9/lib/Doctrine/ORM/QueryBuilder.php#L651